File: original/CS360_App/app/src/main/java/com/example/cs360_app/TargetWeightEntity.java
package com.example.cs360_app;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(
        foreignKeys = @ForeignKey(
                entity = UserEntity.class,
                parentColumns = "id",
                childColumns = "user_id",
                onDelete = ForeignKey.CASCADE
        ),
        indices = @Index(value = {"user_id"}, unique = true)
)
public class TargetWeightEntity {
    @PrimaryKey(autoGenerate = true)
    private int id;

    @ColumnInfo(name = "target_weight")
    private String targetWeight;

    @ColumnInfo(name = "user_id")
    private int userId;

    // == Getters and Setters == //
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getTargetWeight() {
        return targetWeight;
    }

    public void setTargetWeight(String targetWeight) {
        this.targetWeight = targetWeight;
    }
}

File: enhancements/CS499_App/app/src/main/java/com/example/cs499_app/TargetWeightEntity.kt
package com.example.cs499_app

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    foreignKeys = [ForeignKey(
        entity = UserEntity::class,
        parentColumns = ["id"],
        childColumns = ["user_id"],
        onDelete = ForeignKey.CASCADE
    )], indices = [Index(value = ["user_id"], unique = true)]
)
class TargetWeightEntity {
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    @ColumnInfo(name = "target_weight")
    var targetWeight: String? = null

    @ColumnInfo(name = "user_id")
    var userId: Int = 0
}

// TODO: Delete file and test app